Makefile

```diff
@@ -1,26 +1,66 @@
+.PHONY: help venv conda docker docstyle format style black test lint check
+.DEFAULT_GOAL = help
+
+PYTHON = python
+PIP = pip
+CONDA = conda
+SHELL = bash
+
+help:
+	@printf "Usage:\n"
+	@grep -E '^[a-zA-Z_-]+:.*?# .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?# "}; {printf "\033[1;34mmake %-10s\033[0m%s\n", $$1, $$2}'
+
+conda:  # Set up a conda environment for development.
```

Comment:
With a conda env, it may be better to provide an environment file. I'm kind of in two minds about this, though, which is why I have not PR'd my environment spec file to master. Providing an environment spec file is a form of duplication. There's always "minimize duplication" as a principle to adhere to, yet there's also a case for "some duplication can be very enabling". I don't know which is better here.

Reply:
That's a good point: I wonder if its even wise for us to accommodate two different kinds of development environments. If we just standardized on one of them, that might be easier in the long run...?

Also (without trying to start a flame war), if I have a conda environment, does it matter if I `conda install numpy` or if I `pip install numpy`? Is there a meaningful difference? Why is it better to have a separate `env.yml`?

Reply:
With the latest release of conda 4.6, I think it now knows to check pip to satisfy dependencies. Prior to 4.6, I think conda would just clobber over pip, basically, if it didn't see a conda-installed package as a dependency of another package.

Reply:
> Why is it better to have a separate `env.yml`? [also `environment.yml`]

The primary reason is that the `environment.yml` file is akin to a Dockerfile: it's a version-controlled specification for how to build a self-contained conda environment.

I also adopted a bunch of bash tooling surrounding the `environment.yml` file. For example, I use `conda-auto-env` to automatically source an environment if it sees an `environment.yml` file in a directory that I navigate to. If the environment doesn't exist, it'll auto-create the environment based on the spec file.

More of my own conda hacks are detailed [on my blog](http://ericmjl.com/blog/2018/12/25/conda-hacks-for-data-science-efficiency/) as well.

Reply:
I'm fairly strongly opinionated that we should not have both, and feel that we should use stick with a `requirements.txt`

The duplication doesn't make things easier and its simple enough to use pip and conda together

Reply:
Are you suggesting we keep the `make conda` rule, but have it install from `requirements.txt` (no `environment.yml`), or are you suggesting we ditch the `make conda` rule altogether?

Reply:
My preference would be to have a `requirements.txt` or an `environment.yml` but not both. I think a `requirements.txt` is preferable because it doesn't require conda. 

Thats my 2 cents :)

Reply:
Agreed! I'll keep the `make conda` rule: it will just install via the `pip` in the conda environment, instead of using `conda install`.